test/app-luatest/gh_7479_bug_fix_uri_lib_test.lua

```diff
@@ -0,0 +1,90 @@
+    -- Also it contains logic of tarantoolctl connect launch to test instance.
+    local remote_code = [==[
+        local cmd_test_instance = {
+            arg[-1],
+            '-e',
+            [[
+                box.cfg{listen = 0}
+                box.schema.user.grant('guest', 'read, write, execute', 'universe')
+            ]],
+        }
+
+        -- Test instance will be started in a separate process.
+        local ph_test_instance = require('popen').new(cmd_test_instance, {
+            stdin  = 'devnull',
+            stdout = 'devnull',
+            stderr = require('popen').opts.PIPE,
+        })
+        os.execute("sleep " .. tonumber(1))
+
+        -- In /tmp/port-gh-7479.txt we will save auto-assigned port.
+        local file, err = io.open('/tmp/port-gh-7479.txt', 'w')
+        if file == nil then
+            error(err)
+        end
+        -- Get test instance init message that contains port information.
+        local test_instance_init_msg = ph_test_instance:read({stderr = true}):rstrip()
+        -- Get substring like *bound to [::]:49803* in instance init message.
+        local port = string.match(test_instance_init_msg, "[::]:]:%s*(%S+)")
+        if port == nil then
+            -- Get substring like *bound to 0.0.0.0:49803* in instance init message.
+            port = string.match(test_instance_init_msg, "0.0.0.0:%s*(%S+)")
+        end
+        file:write(port)
+        os.execute("sleep " .. tonumber(1))
+
+        local cmd_tarantoolctl = {
+            ']==] .. tarantoolctl_location .. [==[',
+            'connect',
+            -- Indicate only port, without localhost:* before the port value.
+            port,
+        }
+        -- Connection will be established in a separate process.
+        local ph_tarantoolctl = require('popen').new(cmd_tarantoolctl, {
+            stdin  = 'devnull',
+            stdout = 'devnull',
+        })
+        os.execute("sleep " .. tonumber(1))
+        ph_test_instance:close()
+        ph_tarantoolctl:close()
+    ]==]
+
+    local file_remote, err = io.open('/tmp/remote.lua', 'w')
+    if file_remote == nil then
+        error(err)
+    end
+    file_remote:write(remote_code)
+    file_remote:close()
+
+    io.popen('rm *.xlog *.snap &>/dev/null')
+    -- Stderr should contain a message about successful connection.
+    local result = io.popen('tarantool /tmp/remote.lua 2>&1'):read()
+    os.execute("sleep " .. tonumber(1))
+
+    local file_port, err = io.open('/tmp/port-gh-7479.txt', 'r')
+    if file_port == nil then
+        error(err)
+    end
+    local port = file_port:read()
+    file_port:close()
+
+    if port == nil then
+        error('probably a problem with parsing the port value')
+    end
+    t.assert_equals(result, 'connected to localhost:' .. port)
+end
+
+-- Checks whether it is possible to specify only port for net.box.connect.
+g.test_net_box_connect_with_only_port_indication = function()
+    local ok, msg = pcall(net_box.connect, nil, 3301)
```

Comment:
Please don't connect to 3301 in a test - the port may actually be open on the test machine, in which case the test would hang.

Reply:
Done